cmd/utils/executor.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2021 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"runtime"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/manifoldco/promptui"
+	"github.com/manifoldco/promptui/screenbuf"
+	"github.com/okteto/okteto/pkg/log"
+	"golang.org/x/term"
+)
+
+type ManifestExecutor interface {
+	Execute(command string, env []string) error
+}
+
+type Executor struct {
+	outputMode string
+}
+
+type jsonMessage struct {
+	Level     string `json:"level"`
+	Stage     string `json:"stage"`
+	Message   string `json:"message"`
+	Timestamp int64  `json:"timestamp"`
+}
+
+// NewExecutor returns a new executor
+func NewExecutor(output string) *Executor {
+	if output == "tty" && !IsSupportForTTY() {
+		output = "plain"
+	}
+	return &Executor{
+		outputMode: output,
+	}
+}
+
+func IsSupportForTTY() bool {
+	if runtime.GOOS != "windows" {
+		return isSupportForTTY()
+	}
+	return false
+}
+
+// Execute executes the specified command adding `env` to the execution environment
+func (e *Executor) Execute(command string, env []string) error {
```

Comment:
Would it make sense to create different executors depending on the output? I think it would make the logic on each executor cleaner that having several ifs in the execute code. They would have a common interface but some auxiliar func like `displayTTYOutput` would be only present for the TTY executor.

When a new executor is requested, we can create the one according to the output mode specified by the user